src/services/auth.js

```diff
@@ -14,11 +14,27 @@ async function login(data) {
     asyncStore.save("user", JSON.stringify(credentials));
     return credentials;
   } catch (err) {
-    console.log("auth login POST error: ", err.response || err);
+    console.log("auth login error: ", err.response || err);
+    return err;
+  }
+}
+
+async function logout() {
+  let user;
+  try {
+    user = JSON.parse(await asyncStore.retrieve("user"));
+    if (!user) throw new Error("No user to log out.");
```

Comment:
I'm not sure what we need to handle this error. If we are logging out and we don't already have a user, then we could just do nothing, right? If the point is to make sure there is nothing in async storage, we've already accomplished that somehow.

Relatedly, I'm not sure why we need to return the user that we are logging out with. Is it possible that we can just clear async storage if there is a user, do nothing if there isn't one, and then move on?

Reply:
Sorry, didn't catch this when you first wrote it. I wrote a lot of error-throwing mechanisms into the login/logout partly because asyncStore throws a not-very-useful error if it deletes something that doesn't exist, and partly so that we're alerted if anything unusual happens-- e.g. this error should never happen, but if it does, this ensures that we know about it and it doesn't fail silently, because the logout functionality should not be available to a user unless there is login information.